terraform/modules/vpc/endpoints.tf

```diff
@@ -0,0 +1,17 @@
+resource "aws_vpc_endpoint" "concourse" {
+  count             = local.zone_count
+  vpc_id            = module.vpc.vpc.id
+  service_name      = "aws.concourse.endpoint"
+  vpc_endpoint_type = "Interface"
+
+  security_group_ids = [var.loadbalancer.security_group_id]
+
+  subnet_ids          = [aws_subnet.private[count.index].id]
+  private_dns_enabled = true
+}
+
+
+resource "aws_vpc_endpoint_service" "example" {
```

Comment:
Reading the docs, I think this service needs to sit in the internet-egress VPC.  And this is where my confusion came from. Minor nit: `example` -> `internet_proxy`

Basically, this is the service that sits in front of the Internet proxy LBs.  Then the VPC endpoints you define in the consumers can attach to it. This makes more sense in the context of the DNS discussion at https://docs.aws.amazon.com/vpc/latest/userguide/endpoint-service.html#vpc-service-private-dns I think.